test/jasmine/tests/select_test.js

```diff
@@ -545,108 +584,70 @@ describe('select box and lasso', function() {
         }
 
         function assertRanges(expected) {
-            var ranges = (eventData.range || {}).mapbox || [];
+            var ranges = (selectedData.range || {}).mapbox || [];
             expect(ranges).toBeCloseTo2DArray(expected, 1, 'select box range (in lon/lat)');
         }
 
         function assertLassoPoints(expected) {
-            var lassoPoints = (eventData.lassoPoints || {}).mapbox || [];
+            var lassoPoints = (selectedData.lassoPoints || {}).mapbox || [];
             expect(lassoPoints).toBeCloseTo2DArray(expected, 1, 'lasso points (in lon/lat)');
         }
 
-        Plotly.plot(gd, fig).then(function() {
-            var selectingCnt = 0;
-            var selectedCnt = 0;
-            var deselectCnt = 0;
-
-            gd.once('plotly_selecting', function() {
-                assertSelectionNodes(1, 2);
-                selectingCnt++;
-            });
-
-            gd.once('plotly_selected', function(d) {
-                assertSelectionNodes(0, 2);
-                selectedCnt++;
-                eventData = d;
-            });
-
-            gd.once('plotly_deselect', function() {
-                deselectCnt++;
-                assertSelectionNodes(0, 0);
-            });
+        function checkEventCounts(selecting, selected, deselect, msg) {
+            expect(selectingCnt).toBe(selecting, 'plotly_selecting call count: ' + msg);
+            expect(selectedCnt).toBe(selected, 'plotly_selected call count: ' + msg);
+            expect(deselectCnt).toBe(deselect, 'plotly_deselect call count: ' + msg);
+        }
 
+        Plotly.plot(gd, fig).then(function() {
+            resetEvents(gd);
             drag([[370, 120], [500, 200]]);
+            return selectedPromise;
+        })
+        .then(function() {
             assertPoints([{lon: 30, lat: 30}]);
             assertRanges([[21.99, 34.55], [38.14, 25.98]]);
+            checkEventCounts(1, 1, 0, 'select');
 
-            return doubleClick(250, 200).then(function() {
-                expect(selectingCnt).toBe(1, 'plotly_selecting call count');
-                expect(selectedCnt).toBe(1, 'plotly_selected call count');
-                expect(deselectCnt).toBe(1, 'plotly_deselect call count');
-            });
+            return doubleClick(250, 200);
+        })
+        .then(deselectPromise)
+        .then(function() {
+            // deselect gives an extra plotly_selected event on the first click
+            // event data is undefined
+            checkEventCounts(1, 2, 1, 'deselect');
```

Comment:
The `gd.once` we were using in this test before hid this behavior - which exists on master as well as in this branch. I imagine we would prefer to not emit these events with undefined data - which happen I guess when we get [here](https://github.com/plotly/plotly.js/blob/80508e8a60698459f01375c1aa90a8a7e52e0096/src/plots/cartesian/select.js#L225) but `dragged === false`. @etpinard shall I remove that extra event and put this test back as it appeared to be previously?

Reply:
>  I imagine we would prefer to not emit these events with undefined data

Hmm. I'm thinking some users might find it useful to have `plotly_selected` fired even when the selection polygon is smaller than the minimum allowed. In this case, `plotly_selected` essentially mimics `plotly_deselect` - which some users might not know about. 

So, I'd vote for keeping the current behavior and updating the test (good catch by the way :medal_sports: )

But, maybe this is something we might want to change in v2, especially now that click selections are on the table for future development.